tensorflow/python/keras/benchmark/benchmark_util.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Common utils for benchmark."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import timeit
+import numpy as np
+
+from tensorflow.python.keras import callbacks
+from tensorflow.python.keras.benchmark import distribution_util
+
+
+class TimerCallBack(callbacks.Callback):
+  """Callback for logging time in each epoch or batch."""
+
+  def __init__(self):
+    self.times = []
+    self.timer = timeit.default_timer
+    self.startup_time = timeit.default_timer()
+    self.recorded_startup = False
+
+  def on_epoch_begin(self, e, logs):
+    self.epoch_start_time = self.timer()
+
+  def on_epoch_end(self, e, logs):
+    self.times.append(self.timer() - self.epoch_start_time)
+
+  def on_batch_end(self, e, logs):
+    if not self.recorded_startup:
+      self.startup_time = self.timer() - self.startup_time
+      self.recorded_startup = True
+
+
+def measure_performance(model_fn,
+    x=None,
+    y=None,
+    epoch=2,
+    batch_size=32,
+    run_iters=4,
+    optimizer=None,
+    loss=None,
+    metrics=None,
+    verbose=0,
+    num_gpus=0,
+    distribution_strategy='off'):
+  """ Run models and measure the performance.
+
+  Arguments:
+    model_fn: Model built by user.
+    x: Input data. Keep the same definition with keras.fit.
```

Comment:
nit: x: Input data. See \`x\` in the \`fit()\` method of \`keras.Model\`. 

With \`keras.Model\`, it will directly link to the public API of `tf.keras.Model`. So if you want to refer some public API , you may follow this format. Check the [Conv1D layer](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/convolutional.py#L444-L462) as an example.

Reply:
Sounds Great, I will write it in that format.